### PR TITLE
Feature #97846 Updated Ccavenue Plugin

### PR DESCRIPTION
**Pull request by tushar33**

### CODE REVIEW

**Comment by parthlawate** [comment]
Good Job @tushar33  !

File: ccavenue/ccavenue.php [hunk 1]
```diff
@@ -135,111 +228,195 @@ function leftshift($str , $num)
...
+
+		// CHECK :compare response order id and send order id in notify URL
 		$trxnstatus = '';
 		$res_orderid = '';
-		$res_orderid = $data['Order_Id'];
-		if($isValid ) {
-			if(!empty($vars) && $res_orderid != $vars->order_id )
+		$res_orderid = $decrypted_data['order_id'];
+
+		if ($isValid)
```
Comment (coolbung) [inline ccavenue/ccavenue.php]: `isValid` seems to be always true, what is the use for it ?

File: ccavenue/ccavenue/Crypto.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
+<?php
```
Comment (coolbung) [inline ccavenue/ccavenue/Crypto.php]: Where did you receive this file from ?

File: ccavenue/ccavenue/helper.php [hunk 3]
```diff
@@ -1,24 +1,56 @@
...
+	/**
+	 * Get ccavenue url
+	 *
+	 * @param   integer  $secure  true/false
+	 *
+	 * @return  url
+	 *
+	 * @since   1.0
+	 */
+	public function buildCcavenueUrl($secure = true)
```
Comment (coolbung) [inline ccavenue/ccavenue/helper.php]: Does sandbox also use the same URL ? 

We should also support sandbox mode.

File: ccavenue/ccavenue.php [hunk 4]
```diff
@@ -1,103 +1,198 @@
...
-		$core_file 	= dirname(__FILE__) . '/' . $this->_name . '/' . 'tmpl' . '/' . $layout.'.php';
-		$override		= JPATH_BASE . '/' . 'templates' . '/' . $app->getTemplate() . '/html/plugins/' . $this->_type . '/' . $this->_name . '/' . $layout.'.php';
-		if(JFile::exists($override))
+		if (empty($layout))
+		{
+			$layout = "default";
+		}
+
+		$core_file = dirname(__FILE__) . '/' . $this->_name . '/' . 'tmpl' . '/' . $layout . '.php';
+		$override = JPATH_BASE . '/' . 'templates' . '/' . $app->getTemplate() . '/html/plugins/' .
```
Comment (manojLondhe) [inline ccavenue/ccavenue.php]: can you change it  from 
`JPATH_BASE . '/' . 'templates' . '/' . $app->getTemplate() . '/html/plugins/' .`
=>
to 
`JPATH_BASE . '/templates/' . $app->getTemplate() . '/html/plugins/' .`

File: ccavenue/ccavenue/Crypto.php (hunk 2 above)
Comment (tushar33) [inline ccavenue/ccavenue/Crypto.php]: It is available with integration kit.

File: ccavenue/ccavenue.php (hunk 4 above)
Comment (tushar33) [inline ccavenue/ccavenue.php]: Yes, I will

File: ccavenue/ccavenue/helper.php (hunk 3 above)
Comment (tushar33) [inline ccavenue/ccavenue/helper.php]: There is another URL for sandbox and I will test it.

**Pull request merged**